esp-hal/CHANGELOG.md

```diff
@@ -45,17 +45,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Changed the parameters of `Spi::with_pins` to no longer be optional (#2133)
 - Renamed `DummyPin` to `NoPin` and removed all internal logic from it. (#2133)
 - The `NO_PIN` constant has been removed. (#2133)
+- The various `dma_buffer` macros now return the buffers as `&mut [u8]` (#2179)
 
 ### Fixed
 
 - SHA driver can now be safely used in multiple contexts concurrently (#2049)
 - Fixed an issue with DMA transfers potentially not waking up the correct async task (#2065)
 - Fixed an issue with LCD_CAM i8080 where it would send double the clocks in 16bit mode (#2085)
 - Fix i2c embedded-hal transaction (#2028)
-- Fix SPI DMA alternating `write` and `read` for ESP32 and ESP32-S2 (#2131)
 - Fix I2C ending up in a state when only re-creating the peripheral makes it useable again (#2141)
-- Fix `SpiBus::transfer` transferring data twice in some cases (#2159)
 - Fixed UART freezing when using `RcFast` clock source on ESP32-C2/C3 (#2170)
+- Fixed various SPI issues (#2131, #2159, #2179)
```

Comment:
If the change log entry is going to be vague then the PR title should be more descriptive I think.
Otherwise it's hard to know what's fixed and if you're affected or not.

Reply:
I guess I can split the PR up and title it less vaguely but that looks unnecessary busywork. I guess I can restore the changelog entries but at the end it's actually hard to know how wide reaching these changes are, apart from "fixes failing tests".